STYLE_GUIDE.md

```diff
@@ -593,3 +593,72 @@ Single line comments will have exactly one space between the `//` token and the
 // to coalesce multiple tiny arrays
 // into a single bigger array
 ```
+
+## Testing
+
+### Ordering
+
+Tests should be ordered alphabetically (within reason) separated into sections based on system inclusion/exclusion criteria. First, list tests that run on all systems, next list tests which include/exclude one system, and lastly list tests which include/exclude more than one system.
+
+```pony
+actor Main is TestList
+  new create(env: Env) => PonyTest(env, this)
+  new make() => None
+
+  fun tag tests(test: PonyTest) =>
+    // Tests below include all systems and are listed alphabetically
+    test(_TestOnAllSystemsA)
+    test(_TestOnAllSystemsB)
+    test(_TestOnAllSystemsC)
+
+    // Tests below include only windows and are listed alphabetically
+    ifdef windows then
+      test(_TestOnWindowsA)
+      test(_TestOnWindowsB)
+      test(_TestOnWindowsC)
+    end
+
+    // Tests below exclude windows and are listed alphabetically
+    ifdef not windows then
+      test(_TestOnAllSystemsExceptWindowsA)
+      test(_TestOnAllSystemsExceptWindowsB)
+      test(_TestOnAllSystemsExceptWindowsC)
+    end
```

Comment:
This looks fine to be, but I'd like to see the guide specify the order of the `ifdef ` for OS and `not`.

So for example should `ifdef osx` come before `ifdef windows`?

where should "not's" etc be ordered.

I'd like to see a little more specification there.

What the specification is, I don't care. That's all you @rhagenson.